src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -2689,3 +2693,104 @@ export const handleUpdateTwilio = [
   validateTwilioCredentials,
   updateTwilioCredentials,
 ] as ControllerHandler[]
+
+export const handleGetGoLinkSuffix: ControllerHandler<{ formId: string }> = (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as AuthedSessionData).user._id
+
+  // Step 1: Get the form after permission checks
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) => {
+        return AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Read,
+        })
+      })
+      // Step 2: After permission checks, get the GoGov link suffix
+      .andThen(() => {
+        return AdminFormService.getGoLinkSuffix(formId)
+      })
+      .map((goLinkSuffix) => res.status(StatusCodes.OK).json(goLinkSuffix))
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error occurred when getting GoGov link suffix',
+          meta: {
+            action: 'handleGetGoLinkSuffix',
+            ...createReqMeta(req),
+            userId: sessionUserId,
+            formId,
+          },
+          error,
+        })
+        const { errorMessage, statusCode } = mapRouteError(error)
+        return res.status(statusCode).json({ message: errorMessage })
+      })
+  )
+}
+
+export const handleSetGoLinkSuffix: ControllerHandler<
+  { formId: string },
+  unknown,
+  { linkSuffix: string; adminEmail: string }
+> = (req, res) => {
+  const { formId } = req.params
+  const { linkSuffix, adminEmail } = req.body
+  const sessionUserId = (req.session as AuthedSessionData).user._id
+
+  // Step 1: Get the form after permission checks
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) => {
+        return AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Read,
```

Comment:
Hm okay, we can change this to write instead! Was originally thinking read permissions should be enough to allow as read collaborators should be able to share the form too. But since the functionality for our gogov integration is so limited, yes we should only allow changes when collaborators have write permissions.

Reply:
Done in e607d873